jasmine/build/bundle.js

```diff
@@ -80,71 +89,91 @@ class Index {
     this.all = [];
     this.docNum = {};
   }
+
 /**
  * A method that retuns a created index
  * @param {string} filename is the name of the file to me indexed
-  * @return {Object} Returns an Object containing the created Index.
+ * @return {Object} Returns an Object containing the created Index.
  */
   getIndex(filename) {
     return this.index[filename];
   }
-  /**
+
+/**
+ * Search
+ *
  * A method that searches for words in an index
+ *
  * @param {string} filename is the name of the file to be searched
  * @param {string} term is what to search for
-  * @return {Object} Returns an Object of the search result.
+ * @return {Object} Returns an Object of the search result.
  */
-  search(filename, ...term) {
+  search(filename, ...terms) {
+    let result = {};
     if (!filename.includes('json')) {
-      term.push(filename);
-      Object.keys(this.index).forEach((key) => {
-        filename = key;
-      });
+      terms.push(filename);
+      result = this.searchAll(terms);
     } else {
       filename = filename.replace(/\.json|\.|\s/g, '');
+      terms = terms.toString().toLowerCase().match(/\w+/g);
```

Comment:
### Critical

Lines `118 - 125` and `140, 142 - 148` have essentially the same lines of code. There's really no need for a `searchAll` method, that logic can and should be in the `search` method, however the duplicate code highlighted for lines `118 - 125` should be abstracted into it's own method and called for the 2 different instances.